### ISSUE TITLE
Removing event handler in javascript client

### ISSUE THREAD

**Issue by KamyarP** [aspnet/SignalR#1325]
Is there any way to register a function pointer and be able to remove it from hub connection later?
Right now I'm using SignalR with React and would like to be able to register an event handler in my component's c`tor and unregister it on component unmount. Something like:

var myHandler = function(data) {
    console.log(data);
}
// Register
connection.on("myEvent", myHandler );
// Unregister
connection.off("myEvent", myHandler );

**Comment by KamyarP** [comment]
The problem was the way react behaves when using ES6 classes.
React components using ES6 classes no longer autobind `this` to non React methods. So I had to manually bind `this` to point to component to correctly access the handler function's instance. i.e. in my component's ctor, I added: 
this.myHandler = this.myHandler .bind(this);
Is there a way to use ES6 arrow function in this scenario? If I use the following line to register the handler, can it be unregistered later?
connection.on('myEvent', (message) => { console.log("handler called.")});
